proposal for docs site generation - One site per version with proxy (#3865)
projectcalico/calico: RELEASING.md

### Publishing the candidate release branch

1. Check out to the candidate release branch that is created as per the instructions [here](#creating-a-candidate-release-branch). 

   ```
   git checkout release-vX.Y
   ```

1. In [netlify.toml](netlify.toml), set the `RELEASE_VERSION` environment variable to `vX.Y`.
   
1. Commit and push your changes. For example:

   ```
   git commit -m "build vX.Y candidate"
   ```

> 1. On netlify locate the `calico-candidate` site. Change the branch this site gets deployed to `release-vX.Y` by changing the branch name in `Settings -> Build & Deploy -> Deploy contexts`. 

Author: good catch, I think we can do this and get rid of extra build in the production. I'll check this further and update the doc.

1. On netlify create a new site using the `release-vX.Y` branch (You should at least have write access to this repo for site creation) 

1. Rename the randomly generated site name to follow the same naming convention as other releases (Ex: `calico-VX-Y`).

> 1. Ensure that the site is generated properly by visiting site URL (Ex. https://calico-VX-Y.netlify.app/archive/vX.Y/).  

Reviewer: nitpick, but can you use a lowercase v e.g. `vX.Y` instead of `VX.Y`
```suggestion
1. Ensure that the site is generated properly by visiting site URL (Ex. https://calico-vX-Y.netlify.app/archive/vX.Y/).  
```

Author: changes throughout the doc in the new push

   make release-publish
   ```

1. Merge the PR. 

1. On netlify update `Production branch` in Deploy contexts to `release-vX.Y` in docs.projectcalico.org site settings and trigger the deployment.
This will cause docs.projectcalico.org to be updated (after a few minutes). Validate that everything looks correct.

## Adding the previous release to docs.projectcalico.org/archive

> 1. Create a new site on netlify by linking previous release branch (ex: vY.Z) of this repository. Keep the default build settings. Rename the generated site to follow the same naming convention as other releases.

Reviewer: What do you mean by `linking previous release branch`?

1. Ensure deployment is done by visiting to `<site-name>.netlify.app/archive/vY.Z/`.

> 1. Check out to production branch. This would be same branch from which `docs.projectcalico.org` root docs are generated.

Reviewer: I don't think the average person understands what "root docs" means so maybe this wording will be more clear.
```suggestion
1. Check out to production branch. The production branch is constantly changing, but should be the current `release-vX.Y`.
```
